bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -122,22 +161,16 @@ actual class DefaultScanner internal constructor(
     }
 
     override suspend fun didStopScanning() {
-        centralManagers.forEach { centralManager ->
-            if (centralManager.state == CBCentralManagerStatePoweredOn) {
-                centralManager.stopScan()
-            }
-        }
-        discoveringDelegates.clear()
-        centralManagers.clear()
+        debug(TAG) { "didStopScanning" }
+        getCentralManager().stopScan()
     }
 
     override fun generateEnableSensorsActions(): List<EnableSensorAction> {
         // Trigger Enable Bluetooth popup
         return listOfNotNull(
             if (!bluetoothEnabledMonitor.isServiceEnabled) {
                 suspend {
-                    val options =
-                        mapOf<Any?, Any>(CBCentralManagerOptionShowPowerAlertKey to true)
+                    val options = mapOf<Any?, Any>(CBCentralManagerOptionShowPowerAlertKey to true)
                     CBCentralManager(null, enabledQueue, options)
```

Comment:
Still mystery why it's here?

Reply:
yep. Probably a question for @Daeda88.
Updated this part as well: reuse central manager also here

Reply:
As mentioned in a remark below, its used to trigger a "Enable Bluetooth" prompt if the code requests it. This allows the StateMachine to trigger enable prompts automatically

Reply:
Well, eventually we (I) got it, but we need to put a comment for user, @dasdranagon can you do please?